### PR TITLE
provisioners/ansible: fix galaxy paths on Windows

### PR DESCRIPTION
**Pull request by gildegoma**
:warning: **not ready for merge yet** :construction: 

* [ ] #6763 wrong config prevent `destroy` to happen, **to be fixed!**
* [x] #6793 Galaxy Integration problem with Ansible 2.0+ (`0` code is no longer returned)
* [ ] #6726 `extra_vars` option ignored by `ansible_local`
* [x] #6757 fixed and verified on a Windows host
* [x] #6740 fixed and verified on a Windows host
* [ ] *anything else?*

### CODE REVIEW

File: plugins/provisioners/ansible/provisioner/base.rb [hunk 1]
```diff
@@ -179,15 +179,16 @@ def extra_vars_argument
...
 
-        def get_galaxy_roles_path(basedir)
+        def get_galaxy_roles_path(base_dir)
           if config.galaxy_roles_path
-            File.expand_path(config.galaxy_roles_path, basedir)
+            Helpers::expand_path_in_unix_style(config.galaxy_roles_path, base_dir)
           else
-            File.join(Pathname.new(config.playbook).expand_path(basedir).parent, 'roles')
+            playbook_path = Helpers::expand_path_in_unix_style(config.playbook, base_dir)
+            File.join(Pathname.new(playbook_path).parent, 'roles')
```
Comment (gildegoma) [inline plugins/provisioners/ansible/provisioner/base.rb]: @megheaiulian could you please confirm that this piece of code works fine on Windows? Many thanks in advance :)

Comment (gildegoma) [inline plugins/provisioners/ansible/provisioner/base.rb]: checked and confirmed via https://github.com/mitchellh/vagrant/issues/6757#issuecomment-168940936

File: plugins/provisioners/ansible/helpers.rb [hunk 2]
```diff
@@ -25,6 +25,12 @@ def self.stringify_ansible_playbook_command(env, command)
         shell_command += shell_arg.join(' ')
       end
 
+      def self.expand_path_in_unix_style(path, base_dir)
```
Comment (gildegoma) [inline plugins/provisioners/ansible/helpers.rb]: @sethvargo do prefer to make this helper function available to a larger scope of vagrant code base?

**Comment by pablocrivella** [comment]
+1

**Comment by boyvinall** [comment]
:thumbsup:

**Comment by gildegoma** [comment]
Damn! I didn't intend to integrate the master stuff when taking #6869 (my bad). For the sake of clarity, I merge these first bug fixes, and will work on specific branch for the next bug fixes (as usual ;-)

**Pull request merged**